ui/src/components/song/SongCard.tsx

```diff
@@ -0,0 +1,116 @@
+import { Box, Container, Popover, Stack, Typography } from '@mui/material';
+import { SongCardProps } from '../../types/song';
+import { useState } from 'react';
+
+const SongCard = (props: SongCardProps) => {
+  const { title, tempo, originalKey, themes, artist, year, lyricsPreview, filterData } = props;
+
+  const displayData = [
+    filterData?.display?.themes ?? true,
+    filterData?.display?.tempo ?? true,
+    filterData?.display?.originalKey ?? true,
+    filterData?.display?.year ?? true,
+    filterData?.display?.code ?? true,
+  ];
+  const fieldData = [themes, tempo, originalKey, year];
+  const CardFields = ['Themes', 'Tempo', 'Original Key', 'Year', 'Code'];
+
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+
+  const handlePopoverOpen = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handlePopoverClose = () => {
+    setAnchorEl(null);
+  };
+
+  const open = Boolean(anchorEl);
+
+  const titleStyle = {
```

Comment:
can you add a TODO comment here about using proper theme from MUI, so the next person can remember